r2-core/src/main/java/com/linkedin/r2/transport/http/client/AsyncPoolImpl.java

```diff
@@ -601,6 +623,25 @@ private void create()
       @Override
       public void run(final SimpleCallback callback)
       {
+        boolean shouldIgnore;
+        synchronized (_lock) {
+          // Ignore the object creation if no one is waiting for the object and the pool already has _minSize objects
+          int totalObjects = _checkedOut + _idle.size();
+          shouldIgnore = _waiters.size() == 0 && totalObjects >= _minSize;
+          if (shouldIgnore) {
+            _statsTracker.incrementIgnoredCreation();
+            if (_poolSize >= 1)
+            {
+              _poolSize--;
```

Comment:
The correctness of this line is not immediately clear and we might want to clarify that it is tightly tied to the shouldCreate methd

Reply:
also I'm not super clear why this would go below 1, at first sight it seem that it would be kind of a bug if ti went below.
I'd be curious if it ever happens

Reply:
This tightly coupled condition is very dangerous being so distant from the origin of the update

Reply:
Added a comment to make it clear.  If you closely watch the creation onError code path - this is exactly done there as well.